### ISSUE TITLE
Trace ID support on TimeLock

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#1765]
See internal comments on PDS-50991.

If a timelock client request fails and its request includes a trace ID, it would be useful to have this trace ID on server logs as well. I'm not entirely sure what this would entail in terms of implementation, but it looks like http-remoting has a fair bit of infrastructure to handle this.

**Comment by rhero** [comment]
We definitely want to add something like this, but the question is whether or not we want to take on moving to http-remoting and doing away with our own custom http client.  Going to file a new ticket to track the decision to move to http-remoting.

**Comment by hsaraogi** [comment]
Blocked on #1999.

**Comment by hsaraogi** [comment]
Blocked on #1766.

**Comment by hsaraogi** [comment]
After switching to remoting3, we should consider using the remoting client and get tracing for free.
